crates/icondiffbot2/src/job_processor.rs

```diff
@@ -3,62 +3,66 @@ use crate::{
     table_builder::OutputTableBuilder,
     CONFIG,
 };
-use anyhow::{Context, Result};
+use diffbot_lib::log::error;
 use diffbot_lib::{github::github_types::CheckOutputs, job::types::Job};
 use dmm_tools::dmi::render::{IconRenderer, RenderType};
 use dmm_tools::dmi::State;
 use dreammaker::dmi::StateIndex;
+use eyre::{Context, Result};
 use hashbrown::HashSet;
 use rayon::prelude::*;
 use std::{
-    collections::hash_map::DefaultHasher,
     fs::File,
     hash::{Hash, Hasher},
     io::{BufWriter, Write},
     path::Path,
 };
-use tokio::runtime::Handle;
-use tracing::{info_span, Instrument};
 
 #[tracing::instrument]
-pub fn do_job(job: &Job) -> Result<CheckOutputs> {
-    // TODO: Maybe have jobs just be async?
-    let handle = Handle::try_current()?;
-    handle.block_on(async { handle_changed_files(job).await })
-}
+pub fn do_job(job: Job) -> Result<CheckOutputs> {
+    let handle = actix_web::rt::Runtime::new()?;
 
-#[tracing::instrument]
-pub async fn handle_changed_files(job: &Job) -> Result<CheckOutputs> {
-    job.check_run.mark_started().await?;
+    handle.block_on(async { job.check_run.mark_started().await })?;
 
     let mut map = OutputTableBuilder::new();
 
     for dmi in &job.files {
-        let file = sha_to_iconfile(job, &dmi.filename, status_to_sha(job, &dmi.status)).await?;
+        let file = sha_to_iconfile(&job, &dmi.filename, status_to_sha(&job, &dmi.status))?;
 
-        let j = job.clone();
-        let states = tokio::task::spawn_blocking(move || render(&j, file)).await??;
+        let states = render(&job, file)?;
 
         map.insert(dmi.filename.as_str(), states);
     }
 
-    map.build().instrument(info_span!("Building table")).await
+    map.build()
 }
 
 #[tracing::instrument]
 fn render(
     job: &Job,
     diff: (Option<IconFileWithName>, Option<IconFileWithName>),
-) -> Result<(String, Vec<String>)> {
+) -> Result<(&'static str, Vec<String>)> {
     // TODO: Alphabetize
     // TODO: Test more edge cases
     match diff {
-        (None, None) => unreachable!("Diffing (None, None) makes no sense"),
+        (None, None) => Ok((
+            "UNCHANGED",
```

Comment:
Technically this should be "moved without changes"

Reply:
this condition shouldn't be hit at all, i've simply made it not panic

Reply:
I was wrong about making this unreachable lol, it happens in normal cases of "we moved a file without changes"

Reply:
no, it's impossible to reach now, i've filtered out all the cases that lead to this

Reply:
ChangeType::Renamed isn't actually consistently reported by github, it can give you a moved-without-changes as a ChangeType::Modified